### ISSUE TITLE
Unable to expose port from console program (.Net 6.0)

### ISSUE THREAD

**Issue by atrakic** [dotnet/dotnet-docker#5144]
### Describe the Bug

I have a scenario where I would like to expose metrics from my console programs. 

I have found this sample: https://learn.microsoft.com/en-gb/dotnet/core/diagnostics/metrics-collection#configure-the-example-app-to-use-opentelemetrys-prometheus-exporter which I try to work with, but for some reason keeps failing reaching metrics url from the host (see bellow). 
 

After starting the container, I can reach metrics url inside the container, but not from the host. 

```
$ docker-compose exec -it app curl localhost:9184/metrics
# TYPE content_length_bytes returned_total counter
# UNIT content_length_bytes returned_total bytes returned
# HELP content_length_bytes returned_total bytes
hats-sold_bytes returned_total 32347 1706625928494

# EOF
```

```
$ curl localhost:9184/metrics
curl: (7) Failed to connect to localhost port 9184 after 0 ms: Couldn't connect to server
```

### Steps to Reproduce

```
$ mkdir -p src
$ dotnet new console
$ dotnet add package OpenTelemetry.Exporter.Prometheus.HttpListener --prerelease 
```

```
# compose.yml
services:
  app:
    build: .
    container_name: app
    tty: true
    environment:
      #- ASPNETCORE_ENVIRONMENT=Development
      - ASPNETCORE_URLS=http://*:9184
    ports:
      - 9184:9184
```

```
# Dockerfile
ARG TAG=6.0

FROM mcr.microsoft.com/dotnet/sdk:$TAG AS build
WORKDIR /src

# copy csproj and restore as distinct layers
COPY src/*.csproj .
RUN dotnet restore

# copy and publish app and libraries
COPY ./src .
RUN dotnet publish --no-restore -o /app

# not using aspnet here
FROM mcr.microsoft.com/dotnet/runtime:$TAG as final
WORKDIR /app
EXPOSE 9184
RUN apt-get update && apt-get install -y curl
COPY --from=build /app .
USER $APP_UID
ENTRYPOINT ["./dotnetapp"]
```

### Output of `docker version`

Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:28:49 2023
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.24.0 (122432)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:31:36 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

**Comment by richlander** [comment]
Which webserver are you using? You are configuring ASP.NET Core, yet not using the ASP.NET Core container image.

**Comment by atrakic** [comment]
> Which webserver are you using? You are configuring ASP.NET Core, yet not using the ASP.NET Core container image.

I'm using OT http library as explained in reference link. 
Runtime image is used since this is a console app, and I would like to expose port with it.

**Comment by lbussell** [comment]
@atrakic I was able to reproduce this by following your instructions. I also validated that Prometheus was able to access the metrics when I run the console app locally. Running the app in an `aspnet` container instead of a `runtime` container also makes no difference. There are a few things to note here -

- The `EXPOSE` instruction is just metadata and doesn't affect which ports docker forwards. You still need to pass the `-p 9184:9184` argument if you run the container directly.
- The `ASPNETCORE_URLS` is not going to do anything if you aren't running ASP.NET.

Theoretically there is nothing stopping you from running any kind of http server in a .NET Runtime container. My guess is this is some sort of configuration issue with OpenTelemetry. I'm not an expert in the library so I recommend you file an issue there - https://github.com/open-telemetry/opentelemetry-dotnet
